### DISCUSSION TITLE
No suggestion button while clicking on a commit of the pull request

### DISCUSSION THREAD

**Discussion by Eleven1Liu** [community/community#7128] [Pull Requests]
Hi Github team,

I'm trying to review the latest commit of a pull request,  but there's no `Insert a suggestion` button in the comment field.
Therefore, I have to use ` ```suggestion This is a suggestion.``` ` to add the suggestion.
Is that an expected behavior? Or the users are expected to find the suggestion button through the tab `Files Changed`.
Do you have any suggestions? Thanks!!

## Behavior
**Step 1.  Click on the latest PR** (so we don't have to go through changes in the previous review)
![image](https://user-images.githubusercontent.com/55077129/139782051-82e9f328-7bdb-4d6f-be6b-cbda9dbe28d7.png)

**Step 2.  Start a review and try to add a suggestion.** 
![image](https://user-images.githubusercontent.com/55077129/139780721-fbee1fa8-9319-44f8-8b26-5e687d56bc71.png)

## Feature Request 
A suggestion button for reviewers to add suggestion to the commit of a PR like the one in tab `Files Changed`.

![image](https://user-images.githubusercontent.com/55077129/139782696-d48b3c65-6c10-460b-80f8-f0e6235a9ad0.png)

**Comment by roro-lv** [comment]
+1

**Comment by maross** [comment]
+1

**Comment by jameshelou** [comment]
+1

**Comment by brx-elon** [comment]
+1

**Comment by laurentjacob** [comment]
+1

**Comment by ayberkozgur** [comment]
+1, never understood why this happens but feels like an oversight

**Comment by Farigh** [comment]
+1 this is really annoying when reviewing PRs with many commits

**Comment by mfdebian** [comment]
Thank god I'm not the only one, anyone know a way to fix this? I no longer get the expected bar with the options!

**Comment by ander-murane** [comment]
+1

**Comment by bahar-p** [comment]
+1

**Comment by rauno-tali** [comment]
+1

**Comment by sneakyvv** [comment]
+1

**Comment by richcooper95** [comment]
I'm guessing the reason this doesn't exist is that the 'suggest changes' change can be immediately committed to the PR branch by the author via 'commit suggestion' -- but that might not be possible if the lines you're suggesting changes for don't exist in the overall changes of the PR (i.e. because they've been changed again in a later commit).

I still think a 'suggest changes' at the per-commit level would be very useful in most cases (if the commit history is sane and doesn't overlap changes) rather than disabling it altogether -- a couple of potential ways to allow this might be:
1. Allow 'suggest changes' in commits, but disable the 'commit suggestion' button if the changes were suggested in a specific commit.
2. Allow 'suggest changes' in commtis, but disable the 'commit suggestion' button if the changes can't be applied to the overall PR changes in a separate commit.

EDIT: Just realised the OP was asking specifically about suggesting changes for the latest commit -- that could also be one of the scenarios where 'suggest changes' is allowed!

**Comment by anibalcesar** [comment]
+1

**Comment by huang-weijie** [comment]
+1

**Comment by kreintjes** [comment]
This feature ("just a simple button" ™️) would still be highly appreciated by me and my team. In this case it really feels like just a button, since the feature works perfectly when typing ` ```suggestion` yourself, copy-pasting the old code and then editing it. I use it for a long time now and no issues with it, except that it's cumbersome to have to type and copy-paste it yourself every time. A button for this when placing a comment on a single commit would be much easier. Feels strange this option is only available in the (all) Files changed tab.

We are 1.5 year further since the original suggestion and still no answer from anyone from GitHub at all (as far as I can tell) 😞

**Reply by ayberkozgur** [reply]
So true, I feel like we should have gone with Gitlab instead

**Comment by worblehat** [comment]
+1

**Comment by felixmeziere** [comment]
+1

**Comment by Shannon890** [comment]
+1

**Comment by mauriciogracia** [comment]
The **Add code suggestion button** is only available when you are checking "Files changed" which is the general view of all the commit changes
![image](https://github.com/community/community/assets/2321661/d5175d37-ef59-4491-b2ae-96760be21c44)

Here is the button
![image](https://github.com/community/community/assets/2321661/339fd64d-5637-4ac0-8b88-639897976655)

**Reply by Farigh** [reply]
This is the whole point of this issue : to add this button when filtering on a (subset of) commit(s)

**Reply by mauriciogracia** [reply]
I got here because I could not find that button, I'm just providing a way for anyone else to find it.

Also what if a certain commit made changes to 3 lines of code, and then a next commit remove them, the code suggestion will be obsolete, for me that is the reason is located in the FILES CHANGED and not for every commit

**Reply by DanKaplanSES** [reply]
> I got here because I could not find that button, I'm just providing a way for anyone else to find it.

That's one way to find it, but [it didn't help me](https://stackoverflow.com/questions/77706163/in-github-how-do-i-make-a-suggested-change-on-a-merged-pr-when-i-dont-have-wri). It seems you can't make a suggestion once a PR has been merged. That may be what some people are running into.

As you may be able to tell, I'm much more familiar with Git than GitHub. After a `git merge`, it's perfectly fine to continue working on either branch and merge again. A PR _seems_ similar to a `git merge` but these are separate concepts. Once you've merged a PR, you can't merge the PR a second time. Therefore, from a Git**Hub** perspective, it makes sense for the suggestion button to disappear once you've merged a PR.

That being said, I think it would be a much better user experience if the button were disabled with a helpful tool tip to explain why it can't be used. 

Note: I'm aware this doesn't relate to the OP, but like @mauriciogracia , I found this thread because I googled for why I can't find the suggestion button.

**Reply by YaqiWang** [reply]
@mauriciogracia It will be a good feature if GitHub can detect whether the suggestion button is applicable for a change in a commit. I think a general good coding practice should be that in one pull request, multiple commits do not modify the same line. If GitHub can provide this feature, reviewers can see quickly that a code line is touched by multiple commits.

**Comment by MarkoPaul0** [comment]
Seems like this button was removed between yesterday and today for me. It was very convenient, don't know why they did this.

**Comment by anthony-costar** [comment]
Is the pull request already merged? If so, you can't add a suggestion after the fact. The button should still be there for open pull requests.

**Reply by Yoftahe1** [reply]
❤️

**Comment by sricks** [comment]
This functionality would be useful for commits and already-merged PRs. Yes, the "implement suggestion" won't work, but it's a useful way to show what the requested change is (vs explaining it with sentences). Github should simply remove the "implement" button when it's not relevant.

**Comment by alexhwoods** [comment]
Note, you can just type the suggestion as a code block, e.g.

![CleanShot 2024-10-17 at 11 34 59](https://github.com/user-attachments/assets/300c637d-c084-40b4-8aa5-ef1f738f45e3)

**Reply by alexhwoods** [reply]
That being said, Github, please add the button! It seems so simple

**Comment by Sh-Pe** [comment]
+1

**Comment by DanieleTrapani** [comment]
+1

**Comment by CeballosGene** [comment]
+1

**Comment by Hulkmaster** [comment]
+1

**Comment by naiyoma** [comment]
+1

**Comment by pskowronek** [comment]
+1 :)

**Comment by CoreyAccel** [comment]
Also missing, odd

**Comment by wenhmlittlefish** [comment]
I just found one important point. I can see the "insert suggestion" button on an unmerged Pull request, but if the Pull request has been merged then you can't see the "insert suggestion" button.

**Reply by DanKaplanSES** [reply]
[Correct!](https://github.com/orgs/community/discussions/7128#discussioncomment-7931581)
